src/edu/washington/escience/myria/CsvTupleWriter.java

```diff
@@ -96,12 +97,10 @@ public void error() throws IOException {
     }
   }
 
-  private void writeBBtoFile(final ByteBuffer bb, final String fn) throws IOException {
-    File file = new File(fn);
-    FileOutputStream fos = new FileOutputStream(file, false);
-    FileChannel wChannel = fos.getChannel();
-    wChannel.write(bb);
-    wChannel.close();
-    fos.close();
+  private static String createTempFile(final ByteBuffer bb) throws IOException {
+    Path path = Files.createTempFile("out", null);
+    File file = path.toFile();
```

Comment:
This `file` is unnecessary, you can use `path.toAbsolutePath()` to get the absolute path.